homeassistant/loader.py

```diff
@@ -195,22 +195,45 @@ def __init__(
         hass: "HomeAssistant",
         pkg_path: str,
         file_path: pathlib.Path,
-        manifest: Dict,
+        manifest: Dict[str, Any],
     ):
         """Initialize an integration."""
         self.hass = hass
         self.pkg_path = pkg_path
         self.file_path = file_path
-        self.name: str = manifest["name"]
-        self.domain: str = manifest["domain"]
-        self.dependencies: List[str] = manifest["dependencies"]
-        self.after_dependencies: Optional[List[str]] = manifest.get(
-            "after_dependencies"
-        )
-        self.requirements: List[str] = manifest["requirements"]
-        self.config_flow: bool = manifest.get("config_flow", False)
+        self.manifest = manifest
         _LOGGER.info("Loaded %s from %s", self.domain, pkg_path)
 
+    @property
+    def name(self) -> str:
+        """Return name."""
+        return cast(str, self.manifest["name"])
+
+    @property
+    def domain(self) -> str:
+        """Return domain."""
+        return cast(str, self.manifest["domain"])
+
+    @property
+    def dependencies(self) -> List[str]:
+        """Return dependencies."""
+        return cast(List[str], self.manifest["dependencies"])
+
+    @property
+    def after_dependencies(self) -> Optional[List[str]]:
+        """Return after_dependencies."""
+        return cast(Optional[List[str]], self.manifest.get("after_dependencies"))
+
+    @property
+    def requirements(self) -> List[str]:
```

Comment:
Same as above, it would allow us to remove unneeded/unused elements from the JSON manifest in the end.